ale/kernel_access.py

```diff
@@ -295,3 +298,65 @@ def remove_dups(listofElements):
     else:
         warnings.warn(f"{format_as} is not a valid format, returning as dict")
         return kernels
+
+
+def spiceql_call(function_name = "", function_args = {}, use_web=False):
+    """
+    Interface to SpiceQL (Spice Query Library) for both Offline and Online use
+
+    This function will access the value passed through props defined as `web`. This
+    value determines the access pattern for spice data. When set to Online, you will
+    access the SpiceQL service provided through the USGS Astro AWS platform. This service
+    performs kernel and data aquisition. If set to Offline, you will access locally loaded
+    kernels, and SpiceQL will do no searching for you.
+
+    Parameters
+    ----------
+    functions_name : str
+                        String defineing the function to call, properly exposed SpiceQL
+                        functions should map 1-to-1 with endpoints on the service
+                    
+    function_args : dict
+                    Dictionary of arguments used by the function
+
+    Returns : any
+                Any return from a SpiceQL function
+    """
+    if not use_web:
+        func = getattr(pyspiceql, function_name)
+
+        # Ensure that in offline mode we anticipate the user loading/passing their own kernels
+        # to ALE
+        return func(**function_args)
+
+    url = "https://spiceql-dev.prod-asc.chs.usgs.gov/v1/"
```

Comment:
Even though this is only an incremental step towards turning this on, I would strongly advocate using an ENV variable here to encode. That seems the most likely way a user might want to set / change this. Sure, this might eventually be hard coded, but I would advocate against that since it tightly couples this ALE version to a specific spiceql server version.

Reply:
100% agree, I think providing some way to set it then defaulting might be a good middle ground?